research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+9. Set _topDocument_'s
+   [autofocus processed flag](https://html.spec.whatwg.org/multipage/interaction.html#autofocus-processed-flag)
+   to true.
+
+<p class="note">
+  Without the precense of the autofocus or delegatefocus attribute, focus remains on the active
+  element. This behavior is to enable scenarios where the popup is used in a composite control. For
+  example, a combobox where the user expects their focus to stay in the text input instead of moving
+  automatically to the listbox popup as and when it appears.
+</p>
+
+<p class="question">
+  These focusing steps effectively assert that the delegatesfocus attribute takes precedence over
+  the autofocus attribute. Is that the desired behavior?
+</p>
+
+## Anchoring a `popup` to another element
+
+Many popups have a visual-logical relationship with another element that may not be a part of the
+`popup` element's ancestry chain. For example, a `button` may control a `popup` element, and the
+`popup` needs to be positioned with respect to this `button`. Or the `popup` element may simply be
+used to draw attention to or describe another element. The `anchor` attribute indicates this
+relationship. When specified, the attribute's value must be the ID of an element in the same
+[node document](https://dom.spec.whatwg.org/#concept-node-document) as the `popup` element.
+
+**Example:**
+
+```html
+<button id="myButton">Anchor element</button>
+<popup open anchor="myButton">
+  <p>
+    <strong>New!</strong>
+    I’m some sort of educational UI…
+  </p>
+</popup>
+```
+
+The target of the `anchor` attribute participates in determining whether a candiate _popup_ element
+is "nested" in a currently-shown `popup`; refer to <a href="#hiding-current-popups">hiding
+currently-shown <code>popup</code> elements steps</a>.
+
+<p class="todo">
+  Whenever there is a spec to point to for CSS anchored positioning, call out here that the anchor
+  attribute is used to help resolve CSS <code>anchor()</code> functions.
+</p>
+
+## Hiding a `popup` element
+
+### The `hide()` method
+
+The `hide()` method hides a `popup` element. When `hide()` is invoked, the user agent must run
+[hiding a `popup` element steps](#hiding-popup-steps) from the `popup` element _subject_ on which
+the method was invoked.
+
+### Light dismissal
+
+When a `popup` element is shown and a light dismiss interaction occurs, the user agent must run
+[hiding a `popup` element steps](#hiding-popup-steps) from the `popup` element _subject_.
+
+<p class="todo">
+  Land <a href="https://github.com/openui/open-ui/pull/243">#243</a> and link out to this canonical
+  definition of light dismissal for consistency. Layout changes to the popup or its anchor element
+  are also ground for light dismissal, figure out how to surface this special case. How to handle in
+  prose light dismiss behaviors that may dismiss some but not all currently-shown popups?
+</p>
+
+<h3 id="hiding-popup-steps">
```

Comment:
I'm of two minds about this, because I think what you're saying does make sense. But on the other hand, I've tried to order these sections by popup lifecycle (showing, focusing, anchoring, hiding) and it might read a little funny if those are out of order. The "Hiding other popups" section probably makes that a little less clear, since it is about _hiding_ (end of lifecycle) but is a subtask of showing algos. Thoughts?

Reply:
I ended up moving things around like this: https://github.com/openui/open-ui/pull/355/commits/ecb4964478f3c7330f58914a79955000d906108c